src/GafferScene/Preview/RendererAlgo.cpp

```diff
@@ -718,6 +721,125 @@ void outputObjects( const ScenePlug *scene, const IECore::CompoundObject *global
 	parallelProcessLocations( scene, output );
 }
 
+void applyCameraGlobals( IECore::Camera *camera, const IECore::CompoundObject *globals )
+{
+
+	// apply the resolution, aspect ratio and crop window
+
+	V2i resolution( 640, 480 );
+
+	const V2iData *resolutionOverrideData = camera->parametersData()->member<V2iData>( "resolutionOverride" );
+	if( resolutionOverrideData )
+	{
+		// We allow a parameter on the camera to override the resolution from the globals - this
+		// is useful when defining secondary cameras for doing texture projections.
+		/// \todo Consider how this might fit in as part of a more comprehensive camera setup.
+		/// Perhaps we might actually want a specific Camera subclass for such things?
+		resolution = resolutionOverrideData->readable();
+	}
+	else
+	{
+		if( const V2iData *resolutionData = globals->member<V2iData>( "option:render:resolution" ) )
+		{
+			resolution = resolutionData->readable();
+		}
+
+		if( const FloatData *resolutionMultiplierData = globals->member<FloatData>( "option:render:resolutionMultiplier" ) )
+		{
+			resolution.x = int((float)resolution.x * resolutionMultiplierData->readable());
+			resolution.y = int((float)resolution.y * resolutionMultiplierData->readable());
+		}
+
+		const FloatData *pixelAspectRatioData = globals->member<FloatData>( "option:render:pixelAspectRatio" );
+		if( pixelAspectRatioData )
+		{
+			camera->parameters()["pixelAspectRatio"] = pixelAspectRatioData->copy();
+		}
+	}
+
+	camera->parameters()["resolution"] = new V2iData( resolution );
+
+	// calculate an appropriate screen window
+
+	camera->addStandardParameters();
+
+	// apply overscan
+
+
+	Box2i renderRegion( V2i( 0 ), resolution - V2i( 1 ) );
+
+	const BoolData *overscanData = globals->member<BoolData>( "option:render:overscan" );
+	if( overscanData && overscanData->readable() )
+	{
+		// get offsets for each corner of image (as a multiplier of the image width)
+		V2f minOffset( 0.0 ), maxOffset( 0.0 );
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanLeft" ) )
+		{
+			minOffset.x = overscanValueData->readable();
+		}
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanRight" ) )
+		{
+			maxOffset.x = overscanValueData->readable();
+		}
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanBottom" ) )
+		{
+			minOffset.y = overscanValueData->readable();
+		}
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanTop" ) )
+		{
+			maxOffset.y = overscanValueData->readable();
+		}
+
+		// convert those offsets into pixel values
```

Comment:
Comment carried over from previous implementation seems slightly inaccurate now - we're applying the offsets rather than just converting them.